### PR TITLE
fix: don't use dash for get all executions

### LINKED ISSUE #653: Don't use "-" on get execution when there is no script name provided, use /executions endpoint directly.
**Describe the enhancement you'd like to see**
on get script 

**Additional context**
Add any other context, CLI input/output, etc. about the enhancement here.

### PR DESCRIPTION
**Pull request by nicufk**
## Pull request description 

Closes #653 

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/kubectl-testkube/commands/scripts/get_script_execution.go [hunk 1]
```diff
@@ -20,7 +21,7 @@ func NewGetExecutionCmd() *cobra.Command {
 			// - 2 - scriptName + executionName
 			var scriptID, executionID string
 			if len(args) == 1 {
-				scriptID = "-"
+				scriptID = ""
```
Comment (exu) [inline cmd/kubectl-testkube/commands/scripts/get_script_execution.go:24]: we can ignore script here at all ?

File: cmd/kubectl-testkube/commands/scripts/start.go [hunk 2]
```diff
@@ -47,7 +48,7 @@ func NewStartScriptCmd() *cobra.Command {
 				watchLogs(execution.Id, client)
 			}
 
-			execution, err = client.GetExecution("-", execution.Id)
+			execution, err = client.GetExecution("", execution.Id)
```
Comment (exu) [inline cmd/kubectl-testkube/commands/scripts/start.go:51]: and remove first parameter here as execution.Id is unique for us

File: cmd/kubectl-testkube/commands/scripts/watch.go [hunk 3]
```diff
@@ -21,7 +22,7 @@ func NewWatchExecutionCmd() *cobra.Command {
 			}
 
 			client, _ := GetClient(cmd)
-			execution, err := client.GetExecution("-", executionID)
+			execution, err := client.GetExecution("", executionID)
```
Comment (exu) [inline cmd/kubectl-testkube/commands/scripts/watch.go:25]: here the same (remove first param)

File: internal/app/api/v1/executions.go [hunk 4]
```diff
@@ -193,13 +193,13 @@ func (s TestKubeAPI) ExecutionLogsHandler() fiber.Handler {
...
 		ctx := c.Context()
-		scriptID := c.Params("id", "-")
+		scriptID := c.Params("id", "")
 		executionID := c.Params("executionID")
 
 		var execution testkube.Execution
 		var err error
 
-		if scriptID == "-" {
+		if scriptID == "" {
```
Comment (exu) [inline internal/app/api/v1/executions.go:202]: remove this branch (handle only execution ID)

File: pkg/api/v1/client/direct.go [hunk 5]
```diff
@@ -69,7 +70,12 @@ func (c DirectScriptsAPI) GetScript(id string) (script testkube.Script, err erro
 }
 
 func (c DirectScriptsAPI) GetExecution(scriptID, executionID string) (execution testkube.Execution, err error) {
-	uri := c.getURI("/scripts/%s/executions/%s", scriptID, executionID)
+
+	uri := c.getURI("/executions/%s", executionID)
+
+	if scriptID != "" {
```
Comment (exu) [inline pkg/api/v1/client/direct.go:76]: remove this one handle only executions by ID

File: pkg/api/v1/client/direct.go [hunk 6]
```diff
@@ -85,11 +91,17 @@ func (c DirectScriptsAPI) GetExecution(scriptID, executionID string) (execution
 
 // ListExecutions list all executions for given script name
 func (c DirectScriptsAPI) ListExecutions(scriptID string, limit int, tags []string) (executions testkube.ExecutionsResult, err error) {
-	var uri string
+
+	uri := "/executions"
+
+	if scriptID != "" {
+		uri = fmt.Sprintf("/scripts/%s/executions", scriptID)
```
Comment (exu) [inline pkg/api/v1/client/direct.go:98]: remove this one handle only executions by ID

File: pkg/api/v1/client/proxy.go [hunk 7]
```diff
@@ -83,7 +83,13 @@ func (c ProxyScriptsAPI) GetScript(id string) (script testkube.Script, err error
 }
 
 func (c ProxyScriptsAPI) GetExecution(scriptID, executionID string) (execution testkube.Execution, err error) {
-	uri := c.getURI("/scripts/%s/executions/%s", scriptID, executionID)
+
+	uri := c.getURI("/executions/%s", executionID)
+
+	if scriptID != "" {
+		uri = c.getURI("/scripts/%s/executions/%s", scriptID, executionID)
```
Comment (exu) [inline pkg/api/v1/client/proxy.go:90]: remove

File: pkg/api/v1/client/proxy.go [hunk 8]
```diff
@@ -96,7 +102,13 @@ func (c ProxyScriptsAPI) GetExecution(scriptID, executionID string) (execution t
 
 // ListExecutions list all executions for given script name
 func (c ProxyScriptsAPI) ListExecutions(scriptID string, limit int, tags []string) (executions testkube.ExecutionsResult, err error) {
-	uri := c.getURI("/scripts/%s/executions", scriptID)
+
+	uri := "/executions"
+
+	if scriptID != "" {
+		uri = fmt.Sprintf("/scripts/%s/executions", scriptID)
```
Comment (exu) [inline pkg/api/v1/client/proxy.go:109]: remove

**Review by exu — CHANGES_REQUESTED** [review]

File: cmd/kubectl-testkube/commands/scripts/start.go (hunk 2 above)
Comment (nicufk) [reply cmd/kubectl-testkube/commands/scripts/start.go]: ✅

File: cmd/kubectl-testkube/commands/scripts/watch.go (hunk 3 above)
Comment (nicufk) [reply cmd/kubectl-testkube/commands/scripts/watch.go]: ✅

File: internal/app/api/v1/executions.go (hunk 4 above)
Comment (nicufk) [reply internal/app/api/v1/executions.go:202]: ✅

File: pkg/api/v1/client/direct.go (hunk 5 above)
Comment (nicufk) [reply pkg/api/v1/client/direct.go]: ✅

File: pkg/api/v1/client/direct.go (hunk 6 above)
Comment (nicufk) [reply pkg/api/v1/client/direct.go:94]: @exu Do we want to change list executions command?

File: pkg/api/v1/client/proxy.go (hunk 7 above)
Comment (nicufk) [reply pkg/api/v1/client/proxy.go]: ✅

File: pkg/api/v1/client/proxy.go (hunk 8 above)
Comment (nicufk) [reply pkg/api/v1/client/proxy.go:105]: @exu Do we want to change list executions command?

File: pkg/api/v1/client/direct.go (hunk 6 above)
Comment (exu) [reply pkg/api/v1/client/direct.go:94]: no sorry I've missed this one - it's ok

File: pkg/api/v1/client/proxy.go (hunk 8 above)
Comment (exu) [reply pkg/api/v1/client/proxy.go:105]: good question - you have right we need it

**Review by exu — APPROVED** [review]

**Pull request merged**